clients/cli/tests/command.rs

```diff
@@ -4314,3 +4316,76 @@ async fn compute_budget(test_validator: &TestValidator, payer: &Keypair) {
         run_transfer_test(&config, payer).await;
     }
 }
+
+async fn scaled_ui_amount(test_validator: &TestValidator, payer: &Keypair) {
+    let config = test_config_with_default_signer(test_validator, payer, &spl_token_2022::id());
+
+    // create token with scaled ui amount extension
+    let token = Keypair::new();
+    let token_keypair_file = NamedTempFile::new().unwrap();
+    write_keypair_file(&token, &token_keypair_file).unwrap();
+    let token_pubkey = token.pubkey();
+    let ui_multiplier = 1.0;
+
+    process_test_command(
+        &config,
+        payer,
+        &[
+            "spl-token",
+            CommandName::CreateToken.into(),
+            token_keypair_file.path().to_str().unwrap(),
+            "--ui-multiplier",
+            &ui_multiplier.to_string(),
+        ],
+    )
+    .await
+    .unwrap();
+
+    let account = config.rpc_client.get_account(&token_pubkey).await.unwrap();
+    let test_mint = StateWithExtensionsOwned::<Mint>::unpack(account.data).unwrap();
+    let extension = test_mint.get_extension::<ScaledUiAmountConfig>().unwrap();
+    assert_eq!(f64::from(extension.multiplier), ui_multiplier);
+
+    // update multiplier
+    let new_ui_multiplier = 5.0;
+    let new_ui_multiplier_timestamp = 0;
+
+    process_test_command(
+        &config,
+        payer,
+        &[
+            "spl-token",
+            CommandName::UpdateUiMultiplier.into(),
+            &token_pubkey.to_string(),
+            &new_ui_multiplier.to_string(),
+            &new_ui_multiplier_timestamp.to_string(),
+        ],
+    )
+    .await
+    .unwrap();
+
+    let account = config.rpc_client.get_account(&token_pubkey).await.unwrap();
+    let test_mint = StateWithExtensionsOwned::<Mint>::unpack(account.data).unwrap();
+    let extension = test_mint.get_extension::<ScaledUiAmountConfig>().unwrap();
+    assert_eq!(f64::from(extension.multiplier), new_ui_multiplier);
+    assert_eq!(f64::from(extension.new_multiplier), new_ui_multiplier);
+    assert_eq!(
+        i64::from(extension.new_multiplier_effective_timestamp),
+        new_ui_multiplier_timestamp
+    );
+
+    // disable authority
+    process_test_command(
+        &config,
+        payer,
+        &[
+            "spl-token",
+            CommandName::Authorize.into(),
+            &token_pubkey.to_string(),
+            "scaled-ui-amount",
+            "--disable",
+        ],
+    )
+    .await
+    .unwrap();
```

Comment:
Should this check that the authority is `None` aftewards?

Reply:
Good call! This check was missing for the confidential transfer test, so I added it there too.